manifests/manifests.go

```diff
@@ -25,9 +28,28 @@ type Params struct {
 	Namespace  string
 }
 
-// GenerateManifests generates weave-gitops manifests from a template
-func GenerateManifests(params Params) ([][]byte, error) {
-	templates, err := fs.ReadDir(wegoAppTemplates, wegoManifestsDir)
+// GenerateWegoAppManifests generates weave-gitops manifests from a template
+func GenerateWegoAppManifests(params Params) ([][]byte, error) {
+	wgaManifests, err := processTemplates(params, wegoAppTemplates, wegoManifestsDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed reading templates directory: %w", err)
+	}
+
+	return wgaManifests, nil
+}
+
+// GenerateHelmWatcherManifests generates helm-watcher manifests from embedded template files.
+func GenerateHelmWatcherManifests(params Params) ([][]byte, error) {
+	hwTemplates, err := processTemplates(params, helmWatcherTemplates, helmWatcherManifestsDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed reading templates directory: %w", err)
+	}
+
+	return hwTemplates, nil
+}
+
```

Comment:
is it worth separating out if they are still 1 deployment in the end? might just be adding more confusion

Reply:
I didn't want to do it together in case the ui is ever pulled out and things are separated into their own proper services. I'm not sure if, for example, the ui will be the only one starting this service. It feels like the wrong place to be honest in the first place.